app/master/build.py

```diff
@@ -136,9 +136,14 @@ def allocate_slave(self, slave):
         :type slave: Slave
         """
         self._slaves_allocated.append(slave)
-        self._num_executors_allocated += min(slave.num_executors, self._max_executors_per_slave)
 
-        slave.setup(self.build_id(), project_type_params=self.build_request.build_parameters())
+        slave.setup(
+            self.build_id(),
+            project_type_params=self.build_request.build_parameters(),
+            num_executors_already_allocated=self._num_executors_allocated
```

Comment:
How do you feel about naming the argument that we send to the slave a more assertive name like `build_executor_start_index` -- I think that name explains what the slave will use it for vs. `num_executors_already_allocated` which is kind of a passive fact about existing state but does not indicate why it's being given to the slave.